added initial .travis.yml

krichter722 opened the pull request:
This allows to test every pushed commit on the Travis CI service. The current `HEAD` already [fails to build on Ubuntu 16.04, 17.10 and 18.04](https://travis-ci.org/krichter722/viking/builds/373286054) which is probably caused by a missing dependency which should be checked in `configure`. You can fix the build issue first if you prefer, I'll rebase the PR, then.

rnorris commented:
I'll sort out the build soon (otherwise busy for several days).
'Only' failing in the test side of things (which doesn't always get built - which I guess is why I missed it)

rnorris commented:
Thanks for taking the time to do this, it's always been on my radar to do it but never found the time.

Here's some thoughts about it
ATM the command "/bin/sh -c blah, blah, blah..." is the same for each line.
Would it be better to make this variable itself with an 'env:' statement?

Also a note about why the need to specify more packages 'libgeoclue-2-dev nettle-dev libmapnik-dev' etc... would be useful.
e.g. something like "#NB Further packages need specifying as these are new dependencies for the latest source code, compared to the dependencies listed in the distribution version."

NB libmapnik-dev shouldn't need to be specified for Zestful and beyond, see https://launchpad.net/ubuntu/zesty/+source/viking but if using a common variable for the command it will make it more complicated

krichter722 commented:
> Here's some thoughts about it
ATM the command "/bin/sh -c blah, blah, blah..." is the same for each line.
Would it be better to make this variable itself with an 'env:' statement?

I spent some time trying to get the Travis build working for different distros in parallel (I work a lot with GitLab CI which allows to specify arbitrary Docker images to test on) which is why I proposed the first successful setup with works for all three major Ubuntu distros. There're certainly more intuitive and shorter ways to formulate the Travis configuration. Please feel free to edit this PR.

> Also a note about why the need to specify more packages 'libgeoclue-2-dev nettle-dev libmapnik-dev' etc... would be useful.
e.g. something like "#NB Further packages need specifying as these are new dependencies for the latest source code, compared to the dependencies listed in the distribution version."

I added some comments to the script.

> NB libmapnik-dev shouldn't need to be specified for Zestful and beyond, see https://launchpad.net/ubuntu/zesty/+source/viking but if using a common variable for the command it will make it more complicated

Removed it for 17.10 and 18.04. I'm usually strictly against redundancies, but I agree that here it causes a large amount of complication.

I added a draft for 14.04 which should be working except for a build error in `geoclue` 2.x which needs to be built from source (reported at https://gitlab.freedesktop.org/geoclue/geoclue/merge_requests/1). Imo it should be left in commented out if you care for 14.04 in order to document the investigated build commands.
